packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart

```diff
@@ -161,12 +161,12 @@ class DefaultTextEditingShortcuts extends Shortcuts {
   // uses different modifier keys as the line/word modifer.
   static const Map<ShortcutActivator, Intent> _commonShortcuts = <ShortcutActivator, Intent>{
     // Delete Shortcuts.
-    SingleActivator(LogicalKeyboardKey.backspace): DeleteCharacterIntent(forward: false),
-    SingleActivator(LogicalKeyboardKey.backspace, control: true): DeleteToNextWordBoundaryIntent(forward: false),
-    SingleActivator(LogicalKeyboardKey.backspace, alt: true): DeleteToLineBreakIntent(forward: false),
-    SingleActivator(LogicalKeyboardKey.delete): DeleteCharacterIntent(forward: true),
-    SingleActivator(LogicalKeyboardKey.delete, control: true): DeleteToNextWordBoundaryIntent(forward: true),
-    SingleActivator(LogicalKeyboardKey.delete, alt: true): DeleteToLineBreakIntent(forward: true),
+    SingleActivator(LogicalKeyboardKey.backspace, shift: null): DeleteCharacterIntent(forward: false),
```

Comment:
Why can't you just do this?
```suggestion
    SingleActivator(LogicalKeyboardKey.backspace): DeleteCharacterIntent(forward: false),
    SingleActivator(LogicalKeyboardKey.backspace, shift: true): DeleteCharacterIntent(forward: false),
```

Reply:
I'm concerned that there would be too many activators, both for performance and for maintainability. We're already seeing `shift?` for almost every modifier combination with delete or backspace.

Reply:
But adding a tri-state for these complicates both the internal logic and developer reasoning about it. Have you actually seen a performance problem? Because the map is O(1) for lookup, and creating these entries should be very cheap, I wouldn't expect it to have an impact unless there were thousands of them.